packages/pytest-simcore/src/pytest_simcore/rabbit_service.py

```diff
@@ -39,37 +39,51 @@ async def wait_till_rabbit_responsive(url: str) -> None:
 
 
 @pytest.fixture(scope="function")
-async def rabbit_config(docker_stack: Dict, testing_environ_vars: Dict) -> RabbitConfig:
+async def rabbit_settings(
+    docker_stack: Dict, testing_environ_vars: Dict  # stack is up
+) -> RabbitSettings:
+    """Returns the settings of a rabbit service that is up and responsive"""
+
     prefix = testing_environ_vars["SWARM_STACK_NAME"]
     assert f"{prefix}_rabbit" in docker_stack["services"]
-    rabbit_config = RabbitConfig(
-        user=testing_environ_vars["RABBIT_USER"],
-        password=testing_environ_vars["RABBIT_PASSWORD"],
-        host="127.0.0.1",
-        port=get_service_published_port("rabbit", testing_environ_vars["RABBIT_PORT"]),
-        channels=json.loads(testing_environ_vars["RABBIT_CHANNELS"]),
+
+    port = get_service_published_port("rabbit", testing_environ_vars["RABBIT_PORT"])
+
+    settings = RabbitSettings(
+        RABBIT_USER=testing_environ_vars["RABBIT_USER"],
+        RABBIT_PASSWORD=testing_environ_vars["RABBIT_PASSWORD"],
+        RABBIT_HOST="127.0.0.1",
+        RABBIT_PORT=int(port),
+        RABBIT_CHANNELS=json.loads(testing_environ_vars["RABBIT_CHANNELS"]),
     )
 
-    url = rabbit_config.dsn
-    await wait_till_rabbit_responsive(url)
+    await wait_till_rabbit_responsive(settings.dsn)
 
-    return rabbit_config
+    return settings
 
 
 @pytest.fixture(scope="function")
-async def rabbit_service(rabbit_config: RabbitConfig, monkeypatch) -> RabbitConfig:
-    monkeypatch.setenv("RABBIT_HOST", rabbit_config.host)
-    monkeypatch.setenv("RABBIT_PORT", str(rabbit_config.port))
-    monkeypatch.setenv("RABBIT_USER", rabbit_config.user)
-    monkeypatch.setenv("RABBIT_PASSWORD", rabbit_config.password.get_secret_value())
-    monkeypatch.setenv("RABBIT_CHANNELS", json.dumps(rabbit_config.channels))
+async def rabbit_service(
+    rabbit_settings: RabbitSettings, monkeypatch
+) -> RabbitSettings:
+    """Sets env vars for a rabbit service is up and responsive and returns its settings as well
+
+    NOTE: Use this fixture to setup client app
+    """
+    monkeypatch.setenv("RABBIT_HOST", rabbit_settings.RABBIT_HOST)
+    monkeypatch.setenv("RABBIT_PORT", str(rabbit_settings.RABBIT_PORT))
+    monkeypatch.setenv("RABBIT_USER", rabbit_settings.RABBIT_USER)
+    monkeypatch.setenv(
+        "RABBIT_PASSWORD", rabbit_settings.RABBIT_PASSWORD.get_secret_value()
+    )
+    monkeypatch.setenv("RABBIT_CHANNELS", json.dumps(rabbit_settings.RABBIT_CHANNELS))

```

Comment:
minor: are these monkeypatch necessary? since the rabbit_settings are already setting them? but the port ok.

Reply:
This was the design in place. The way I understood is:
- start stack and wait until is up
- wait until given service is responsive (e.g. redis, rabbit, etc)
- fill settings to create a client for that service
and then, there are two types of fixtures
1. ``*_service`` guarantees services is up and running, and env vars to use it are in monkeypatched
2. ``*_client``  guarantees services is up and running, and returns a client ready to interact with it